vsphere/panels.libsonnet

```diff
@@ -0,0 +1,1118 @@
+local g = import './g.libsonnet';
+local commonlib = import 'common-lib/common/main.libsonnet';
+local utils = commonlib.utils;
+
+{
+  new(this):
+    {
+      local t = this.grafana.targets,
+      local stat = g.panel.stat,
+      local table = g.panel.table,
+      local barGauge = g.panel.barGauge,
+      local fieldOverride = g.panel.table.fieldOverride,
+
+      vmOnStatus:
+        commonlib.panels.generic.stat.info.new(
+          'VMs on',
+          targets=[t.vmOnStatus],
+          description='The number of virtual machines currently powered on.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      vmOffStatus:
+        commonlib.panels.generic.stat.info.new(
+          'VMs off',
+          targets=[t.vmOffStatus],
+          description='The number of virtual machines currently powered off.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      vmSuspendedStatus:
+        commonlib.panels.generic.stat.info.new(
+          'VMs suspended',
+          targets=[t.vmSuspendedStatus],
+          description='The number of virtual machines currently in a suspended state.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      vmTemplateStatus:
+        commonlib.panels.generic.stat.info.new(
+          'VM templates',
+          targets=[t.vmTemplateStatus],
+          description='The number of virtual machine templates available.'
+        )
+        + stat.options.withGraphMode('none'),
+      vmOnStatusCluster:
+        commonlib.panels.generic.stat.info.new(
+          'VMs on',
+          targets=[t.vmOnStatusCluster],
+          description='The number of virtual machines currently powered on within the cluster.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      vmOffStatusCluster:
+        commonlib.panels.generic.stat.info.new(
+          'VMs off',
+          targets=[t.vmOffStatusCluster],
+          description='The number of virtual machines currently powered off within the cluster.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      vmSuspendedStatusCluster:
+        commonlib.panels.generic.stat.info.new(
+          'VMs suspended',
+          targets=[t.vmSuspendedStatusCluster],
+          description='The number of virtual machines currently in a suspended state within the cluster.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      vmTemplateStatusCluster:
+        commonlib.panels.generic.stat.info.new(
+          'VM templates',
+          targets=[t.vmTemplateStatusCluster],
+          description='The number of virtual machine templates available within the cluster.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      clusterCountStatus:
+        commonlib.panels.generic.stat.info.new(
+          'Cluster count',
+          targets=[t.clusterCountStatus],
+          description='The total number of clusters in the vCenter environment.'
```

Comment:
In terms of description, I'd have to refer to @StefanKurek. I believe its fine for the description, I will be removing it from the selector names since I have to reformat those either way.